### PR TITLE
fix: raise a invalid url exception when open link url is not valid

### PR DESCRIPTION
**Pull request by amanjeetsingh150**
## Proposed Changes

1. Escape `?` 
2. Raise invalid URL exception exception to avoid throwing illegal state exceptions

## Testing
- [ ] Local
- [ ] Staging

### CODE REVIEW

File: maestro-client/src/main/java/maestro/utils/StringUtils.kt [hunk 1]
```diff
@@ -12,4 +12,7 @@ object StringUtils {
         }
     }
 
+    fun String.toSanitizedIOSLink(): String {
+        return this.replace("?", "\\?")
```
Comment (amanjeetsingh150) [inline maestro-client/src/main/java/maestro/utils/StringUtils.kt:16]: @artem888 is this correct we need to escape ? when using the xcrun simctl open URL command?

I was getting: 

```
zsh: no matches found: testapp://testpage?testparam={TEST:test}
```

Also do we need to escape something else as well?

File: maestro-ios-driver/src/main/kotlin/util/CommandLineUtils.kt [hunk 2]
```diff
@@ -24,7 +24,7 @@ object CommandLineUtils {
         } else {
             ProcessBuilder(*parts.toTypedArray())
                 .redirectOutput(nullFile)
-                .redirectError(nullFile)
+                .redirectError(ProcessBuilder.Redirect.PIPE)
```
Comment (amanjeetsingh150) [inline maestro-ios-driver/src/main/kotlin/util/CommandLineUtils.kt:27]: The errors were not even redirecting to the process due to which it just throws illegal state exception without any meaningful message

File: maestro-client/src/main/java/maestro/utils/StringUtils.kt (hunk 1 above)
Comment (artem888) [reply maestro-client/src/main/java/maestro/utils/StringUtils.kt:16]: I tested locally , looks like any special character needs to be sanitized, not only `?`.

smth like that should work:

```
fun sanitizeURL(url: String): String {
    val sanitizedURL = StringBuilder()
    for (ch in url) {
        if (ch.isLetterOrDigit() || ch == '.' || ch == '-' || ch == '_' || ch == '*' || ch == '/') {
            sanitizedURL.append(ch)
        } else {
            sanitizedURL.append(URLEncoder.encode(ch.toString(), "UTF-8"))
        }
    }
    return sanitizedURL.toString()
}
```

**Review by artem888 — APPROVED** [review]
LGTM after sanitization logic is modified

**Pull request closed without merging**